packages/web/modules/code-review-post/shared/CodeFile/CodeDiscussionView.tsx

```diff
@@ -55,25 +55,52 @@ export const CodeDiscussionView: React.FC<CodeDiscussionViewProps> = ({
   showDiscussion,
   lineNum,
 }) => {
+  const discussionRef = useRef<HTMLDivElement>(null);
   const { owner } = useContext(PostContext);
   const [showReply, setShowReply] = useState(false);
 
+  useEffect(() => {
+    if (showDiscussion) {
+      discussionRef.current!.classList.add("is-open");
+    }
+  }, [showDiscussion, showReply]);
+
+  useEffect(() => {
+    // we only need maxHeight for the animation, we remove it after the
+    // animation ends for the case the height is grater than 2000px
+    if (showDiscussion) {
+      const id = setTimeout(() => {
+        discussionRef.current!.style.maxHeight = "none";
+      }, 200);
+      return () => clearTimeout(id);
+    }
+  }, [showDiscussion]);
+
   return (
     <>
       <button
         className={badgeClassList(showDiscussion)}
         title={showDiscussion ? COLLAPSE : EXPANDED}
         onClick={() => {
           setShowReply(false);
-          toggleDiscussion();
+          if (showDiscussion) {
+            discussionRef.current!.style.maxHeight = "";
+            discussionRef.current!.classList.remove("is-open");
+            setTimeout(() => {
+              toggleDiscussion();
+            }, 200);
+          } else {
+            toggleDiscussion();
+          }
         }}
       >
         <span className="badge-counter">{question.numReplies}</span>
         <span className="badge-icon">▾</span>
       </button>
       {showDiscussion && (
         <DiscussionContainer
-          className="inner-animate-box is-open"
+          ref={discussionRef}
+          className="inner-animate-box"

```

Comment:
DiscussionContainer only exist in the DOM when showDiscussion is true. Using `className={showDiscussion ? "is-open" : ""}` is like set `className="is-open"` since showDiscussion is always true here, if is-open is set when the element is append to the DOM there is no open animation.

That`s is the reason i use the ref to add 'is-open' after the DiscussionContainer is in the DOM.
```javascript
  useEffect(() => {
    if (showDiscussion) {
      discussionRef.current!.classList.add("is-open");
    }
  }, [showDiscussion, showReply]);
```
and also to remove 'is-open'